File: example/detect/detect.cc
#include <iostream>
#include <vector>

#include "opencv2/opencv.hpp"
#include "yolov4-tiny/yolov4-tiny.h"
#include "timer.h"


int main(int argc, char *argv[]){
    if (argc != 4) {
        std::cerr << "Usage:\n\t./detect <model path> <repeat time> <img>\n";
        return 1;
    }
    std::string model_path(argv[1]);
    std::string repeat(argv[2]);
    std::string img_path(argv[3]);

    Yolov4Tiny net(model_path, 0.5, 0.6, 1280, 960);
    cv::Mat img = cv::imread(img_path);

    std::vector<cv::Rect> result;
    for(int i=0; i<std::stoi(repeat); i++){
        result.clear();
        Timer t;
        net.invoke(img, result);
    }

    for(auto box: result){
        cv::rectangle(img, box, cv::Scalar(255,255,0), 2);
    }
    cv::imwrite("./result.jpg", img);
    return 0;
}

File: example/detect/detect.py
import cv2
import time
import numpy as np

width = 256
weight = f"./CowFace-{width}_6000.weights"
config = f"./CowFace-{width}.cfg"
size = (width,width)

net = cv2.dnn.readNet(weight, config)
model = cv2.dnn_DetectionModel(net)
model.setInputParams(size=size,scale=1./255,swapRB=True)

img = cv2.imread("detect.jpg")
out_img = np.copy(img)

for i in range(0,11):
    start = time.time()
    classes, scores, boxes = model.detect(img, 0.75, 0.4)
    end = time.time()
    print("{:.6f}".format((end - start)*1000))

    for (classid, score, box) in zip(classes, scores, boxes):
        cv2.rectangle(out_img, box, (0,0,255), 2)
        cv2.imwrite("./out.jpg", out_img)
